twisted/twisted
# [10035] Build apidocs inside Read the docs environment.

## Scope

The result is available here https://twisted--1466.org.readthedocs.build/en/1466/

To simplify the release process we want to host the API docs inside read the docs.

I find that the Read the Docs theme is easier to read than the Twisted Trac theme for Sphinx.

This provides support for creating API docs via both the old `bin/admin/build-apidocs` and the Read The Docs build.
In this way we can transition with the next release


## Followup work

* Update the CI workflow to fail on apidocs + narrative docs errors - https://twistedmatrix.com/trac/ticket/10066
* Remove usage of apilinks and replace with intersphinx. -  https://twistedmatrix.com/trac/ticket/10057
* Update Twisted own hosted docs to redirect to read the docs for latest version - In Braid

## How to test

Make sure the old script continues to work `tox -e apidocs`

Check that Sphinx docs now has API docs included in the index this is the before last.

Also check that source links for API work.

```
tox  -e apidocs
firefox docs/_build/index.html
```

Check read the docs

Main https://twisted--1466.org.readthedocs.build/en/1466/ 
API https://twisted--1466.org.readthedocs.build/en/1466/api/index.html

Apilinks https://twisted--1466.org.readthedocs.build/en/1466/core/howto/servers.html  link at "Your protocol handling class will usually subclass twisted.internet.protocol.Protocol. "

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10035
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
